pac4j-core/src/main/java/org/pac4j/core/authorization/generator/FromAttributesAuthorizationGenerator.java

```diff
@@ -1,72 +1,80 @@
 package org.pac4j.core.authorization.generator;
 
+import org.pac4j.core.profile.CommonProfile;
+
 import java.util.Arrays;
 import java.util.Collection;
 import java.util.StringTokenizer;
 
-import org.pac4j.core.profile.CommonProfile;
-
 /**
  * <p>Generate the authorization information by inspecting attributes.</p>
  * <p>The attributes containing the roles separated by the {@link #splitChar} property (can be set through {@link #setSplitChar(String)}) are
  * defined in the constructor. It's the same for the attributes containing the permissions.</p>
- * 
+ *
  * @author Jerome Leleu
  * @since 1.5.0
  */
 public class FromAttributesAuthorizationGenerator<U extends CommonProfile> implements AuthorizationGenerator<U> {
-    
-    private final String[] roleAttributes;
-    
-    private final String[] permissionAttributes;
-    
+
+    private final Iterable<String> roleAttributes;
```

Comment:
Maybe we could have `Collection` everywhere, as in the other class? I'm not so clear about the difference between using `Iterable` or `Collection` in this context... I feel like `Collection` is the main way to represent a collection of something, `Iterable` seems too low level in a way. Not sure :)